src/hooks/abilities/utils/getEditAbilities.ts

```diff
@@ -0,0 +1,66 @@
+import { P, match } from 'ts-pattern'
+
+import type { useBasicName } from '@app/hooks/useBasicName'
+
+type BasicName = ReturnType<typeof useBasicName>
+
+export const getEditAbilities = ({
+  address,
+  basicNameData,
+  hasAuthorisedResolver,
+}: {
+  address?: string
+  basicNameData: BasicName
+  hasAuthorisedResolver?: boolean
+}) => {
+  return match([basicNameData])
+    .with(
+      [
+        {
+          ownerData: {
+            ownershipLevel: 'nameWrapper',
+            owner: P.when((owner) => owner === address),
+          },
+          wrapperData: {
+            child: P.select('fuses'),
+          },
+        },
+      ],
+      ({ fuses }) => {
+        return {
+          canEdit: true,
+          canEditRecords: hasAuthorisedResolver,
+          canEditResolver: !fuses.CANNOT_SET_RESOLVER,
+          canEditPermissions: !fuses.CANNOT_BURN_FUSES,
+          canCreateSubdomains: !fuses.CANNOT_CREATE_SUBDOMAIN,
+          canEditTTL: !fuses.CANNOT_SET_TTL,
+        }
+      },
+    )
+    .with(
+      [
+        {
+          ownerData: {
+            ownershipLevel: P.not('nameWrapper'),
```

Comment:
this can be removed right? it's already implied

Reply:
I personally would prefer to leave it there so it makes the declarative case more clear.